### ISSUE TITLE
Is there any training in Cycle Diffusion

### ISSUE THREAD

**Issue by pseudo-usama** [ChenWu98/cycle-diffusion#20]
As we are training two diffusion models independently for two related domains, I've been considering whether there are any specific training techniques involved in the Cycle diffusion process?
Is there any loss function or weight update in Cycle Diffusion process?

If such techniques exist, could you please direct me to the section of your paper where they are explained?
Thanks!

**Comment by ChenWu98** [comment]
Thanks for the question! CycleDiffusion is a training-free method that does not explicitly encourage the alignment between domains. The two diffusion models are trained by running the same training code on two datasets independently.

**Comment by pseudo-usama** [comment]
Can you please explain the process of Cycle Diffusion in a few steps. I'm struggling to grasp the intricacies of the process and would greatly appreciate any guidance you can offer. I would appreciate any response from you.

Thank you for your previous response!

**Comment by ChenWu98** [comment]
I implemented [Cycle Diffusion](https://huggingface.co/docs/diffusers/api/pipelines/cycle_diffusion) in the Hugging Face diffusers library, which is equivalent to (but much more readable than) this repo. Cycle Diffusion can be more easily understood if you know SDEdit. Same as SDEdit, we first sample a noisy image by adding noises to the original image. Then, similar to SDEdit, we denoise this noisy image iteratively. In each iteration, the key difference between Cycle Diffusion and SDEdit is: in SDEdit, the random noise used by the noise_scheduler is randomly sampled from the standard Gaussian distribution; while in Cycle Diffusion, we infer the random noise that can guide the process to the original image conditioned on the original prompt.

**Comment by pseudo-usama** [comment]
Thanks for your kind answer,
But SDEdit is based on `SDE diffusion models`. But I thought you work is based on simple `Stochastic DPM`?
Are these two things same or what am I missing?

**Comment by ChenWu98** [comment]
Yes, Stochastic DPM can be formulated as SDE, and [this paper](https://arxiv.org/abs/2011.13456) provides some details.

**Comment by pseudo-usama** [comment]
Thanks Wu,
Currently, I am attempting to reproduce your work on latent diffusion. I was wondering if there is any possibility that you would be interested in providing guidance and supervision. The purpose of this replication is to serve as the focus of my MS thesis.

Collaborating with you would be a great pleasure for me.
